redaxo/src/addons/structure/pages/index.php

```diff
@@ -5,77 +5,33 @@
  */
 
 $addon = rex_addon::get('structure');
-
-// basic request vars
-$category_id = rex_request('category_id', 'int');
-$article_id = rex_request('article_id', 'int');
-$clang = rex_request('clang', 'int');
-$ctype = rex_request('ctype', 'int');
-
-// additional request vars
-$artstart = rex_request('artstart', 'int');
-$catstart = rex_request('catstart', 'int');
-$edit_id = rex_request('edit_id', 'int');
-$function = rex_request('function', 'string');
-
-$info = '';
-$warning = '';
-
-$category_id = rex_category::get($category_id) ? $category_id : 0;
-$article_id = rex_article::get($article_id) ? $article_id : 0;
-$clang = rex_clang::exists($clang) ? $clang : rex_clang::getStartId();
-
-// --------------------------------------------- Mountpoints
-
-$mountpoints = rex::getUser()->getComplexPerm('structure')->getMountpoints();
-if (1 == count($mountpoints) && 0 == $category_id) {
-    // Nur ein Mointpoint -> Sprung in die Kategory
-    $category_id = current($mountpoints);
-}
-
-// --------------------------------------------- Rechte prüfen
-$KATPERM = rex::getUser()->getComplexPerm('structure')->hasCategoryPerm($category_id);
-
-$stop = false;
-if (rex_clang::count() > 1) {
-    if (!rex::getUser()->getComplexPerm('clang')->hasPerm($clang)) {
-        $stop = true;
-        foreach (rex_clang::getAllIds() as $key) {
-            if (rex::getUser()->getComplexPerm('clang')->hasPerm($key)) {
-                $clang = $key;
-                $stop = false;
-                break;
-            }
-        }
-
-        if ($stop) {
-            echo rex_view::error('You have no permission to this area');
-            exit;
-        }
-    }
-} else {
-    $clang = rex_clang::getStartId();
-}
-
-$context = new rex_context([
-    'page' => 'structure',
-    'category_id' => $category_id,
-    'article_id' => $article_id,
-    'clang' => $clang,
+$structureContext = new rex_structure_context([
+    'category_id' => rex_request('category_id', 'int'),
+    'article_id' => rex_request('article_id', 'int'),
+    'clang_id' => rex_request('clang', 'int'),
+    'ctype_id' => rex_request('ctype', 'int'),
+    'artstart' => rex_request('artstart', 'int'),
+    'catstart' => rex_request('catstart', 'int'),
+    'edit_id' => rex_request('edit_id', 'int'),
+    'function' => rex_request('function', 'string'),
+    'rows_per_page' => 30,
 ]);
 
 // --------------------- Extension Point
 echo rex_extension::registerPoint(new rex_extension_point('PAGE_STRUCTURE_HEADER_PRE', '', [
-    'context' => $context,
+    'context' => $structureContext->getContext(),
```

Comment:
Vllt einfacher mit

$context = $structureContext->getContext();

Dann musst du nicht alle stellen anpassen

Reply:
Das würde ich erstmal so lassen, da die meisten Nutzungen im Code ja im nächsten Schritt in Fragments verschoben werden sollen.